pkg/azuredisk/controllerserver.go

```diff
@@ -581,22 +581,25 @@ func (d *Driver) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (
 		d := disks[i]
 		nodeList := []string{}
 
-		if d.ManagedBy != nil {
-			attachedNode, err := vmSet.GetNodeNameByProviderID(*d.ManagedBy)
-			if err != nil {
-				return nil, err
+		// HyperVGeneration property is only setup for os disks. Only the non os disks should be included in the list
+		if !(d.DiskProperties != nil && d.DiskProperties.HyperVGeneration != "") {
```

Comment:
this is more clear:
`d.DiskProperties == nil || d.DiskProperties.HyperVGeneration == ""`